Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -890,6 +890,26 @@ void TextEditor::keydown_event(KeyEvent& event)
         try_update_autocomplete();
     } };
 
+    if (is_multi_line() && !event.shift() && !event.alt() && event.ctrl() && event.key() == KeyCode::Key_Return) {
+        if (!is_editable())
+            return;
+
+        size_t indent_length = current_line().leading_spaces();
+        DeprecatedString indent = current_line().to_utf8().substring(0, indent_length);
+        execute<InsertLineCommand>(m_cursor, move(indent), InsertLineCommand::InsertPosition::Below);
+        return;
+    }
+
+    if (is_multi_line() && !event.alt() && event.ctrl() && event.shift() && event.key() == KeyCode::Key_Return) {
+        if (!is_editable())
+            return;
+
+        size_t indent_length = current_line().leading_spaces();
+        DeprecatedString indent = current_line().to_utf8().substring(0, indent_length);
+        execute<InsertLineCommand>(m_cursor, move(indent), InsertLineCommand::InsertPosition::Above);
+        return;
+    }
```

Comment:
The only difference I see between these two is whether they pass `::Above` or `::Below` to the command. You could reduce the duplication by combining them together, something like this:

```suggestion
    if (is_multi_line() && !event.alt() && event.ctrl() && event.key() == KeyCode::Key_Return) {
        if (!is_editable())
            return;

        size_t indent_length = current_line().leading_spaces();
        DeprecatedString indent = current_line().to_utf8().substring(0, indent_length);
        if (event.shift()) {
            execute<InsertLineCommand>(m_cursor, move(indent), InsertLineCommand::InsertPosition::Above);
        } else {
            execute<InsertLineCommand>(m_cursor, move(indent), InsertLineCommand::InsertPosition::Below);
        }
        return;
    }
```

(Or reduce it even further with one call to `execute()` if you want.)

Reply:
Thank you for the review, I just merged the two `if` branches into one as your request.
Well, I think I should write code carefully, another PR from me also contains some repetition.